plugins/kubernetesapi-report/src/test/java/com/freenow/sauron/plugins/KubernetesContainersReaderTest.java

```diff
@@ -0,0 +1,100 @@
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1Container;
+import io.kubernetes.client.openapi.models.V1PodSpec;
+import io.kubernetes.client.openapi.models.V1PodTemplateSpec;
+import io.kubernetes.client.openapi.models.V1Probe;
+import java.util.List;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import static com.freenow.sauron.plugins.utils.KubernetesResources.DEPLOYMENT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class KubernetesContainersReaderTest {
+    private static final String SERVICE_LABEL = "label/service.name";
+    private static final String SERVICE_NAME = "serviceName";
+    public static final String PRODUCTION_READINESS = "productionReadiness";
+
+    @Mock
+    private ApiClient apiClient;
+    @Mock
+    private KubernetesGetDeploymentSpecCommand kubernetesGetDeploymentSpecCommand;
+    @InjectMocks
+    private KubernetesContainersReader kubernetesContainersReader;
+
+    @Before
+    public void setUp() {
+        kubernetesContainersReader = new KubernetesContainersReader(kubernetesGetDeploymentSpecCommand, new RetryConfig());
+    }
+
+    @Test
+    public void read()
+    {
+        final var deploymentData = createDeploymentData(true);
+        final var input = dummyDataSet();
+        when(kubernetesGetDeploymentSpecCommand.getDeployment(SERVICE_LABEL, DEPLOYMENT, SERVICE_NAME, apiClient)).thenReturn(deploymentData);
+
+        kubernetesContainersReader.read(input, SERVICE_LABEL, apiClient);
+        assertNotNull(input);
+        assertEquals("true", input.getStringAdditionalInformation(PRODUCTION_READINESS).get());
+    }
+
+    @Test
+    public void deploymentIsNotReady(){
+        final var deploymentData = createDeploymentData(false);
+        final var input = dummyDataSet();
+        when(kubernetesGetDeploymentSpecCommand.getDeployment(SERVICE_LABEL, DEPLOYMENT, SERVICE_NAME, apiClient)).thenReturn(deploymentData);
+
+        kubernetesContainersReader.read(input, SERVICE_LABEL, apiClient);
+        assertNotNull(input);
+        assertEquals("false", input.getStringAdditionalInformation(PRODUCTION_READINESS).get());
+    }
+
+
+    @Test
+    public void deploymentNotFound_setsNoReadiness() {
+        final var input = dummyDataSet();
+        when(kubernetesGetDeploymentSpecCommand.getDeployment(SERVICE_LABEL, DEPLOYMENT, SERVICE_NAME, apiClient))
+                .thenReturn(Optional.empty());
+
+        kubernetesContainersReader.read(input, SERVICE_LABEL, apiClient);
+
+        assertFalse(input.getStringAdditionalInformation(PRODUCTION_READINESS).isPresent());
+    }
+
```

Comment:
We could also have a unit test to cover `deploymentSpec.getTemplate().getSpec() == null`